scripts/build_renewable_profiles.py

```diff
@@ -278,88 +323,167 @@ def calculate_potential(gid, save_map=None):
-                               **resource)
+    if correction_factor != 1.:
+        logger.info(f'correction_factor is set to {correction_factor}')
+
+    # Calculate corrected CFs for each grid cell
+    capacity_factor = correction_factor * func(
+        capacity_factor=True,
+        show_progress='Compute capacity factors: ',
+        **resource)
+
+    # Create a layout: Distribute capacities based on potentials and CF
+    layoutmatrix = potmatrix * capacity_factor
+
+    # Generate RE profile with the constructed layout
+    profile, capacities = func(
+        # Redindex, stack index and conv. to CSR to be consistent with the format expected by atlite
+        matrix=layoutmatrix.reindex_like(
+            cutout.data).stack(spatial=('y', 'x')).data.tocsr(),
+        index=buses,
+        per_unit=True,
+        return_capacity=True,
+        show_progress='Compute profiles: ',
+        **resource)
 
     p_nom_max_meth = config.get('potential', 'conservative')
 
     if p_nom_max_meth == 'simple':
-        p_nom_max = xr.DataArray(np.asarray(potmatrix.sum(axis=1)).squeeze(), [buses])
+        p_nom_max = as_dense(potmatrix.sum(dim=['y', 'x']))
     elif p_nom_max_meth == 'conservative':
         # p_nom_max has to be calculated for each bus and is the minimal ratio
         # (min over all weather grid cells of the bus region) between the available
         # potential (potmatrix) and the used normalised layout (layoutmatrix /
         # capacities), so we would like to calculate i.e. potmatrix / (layoutmatrix /
         # capacities). Since layoutmatrix = potmatrix * capacity_factor, this
         # corresponds to capacities/max(capacity factor in the voronoi cell)
-        p_nom_max = xr.DataArray([1./np.max(capacity_factor[inds]) if len(inds) else 0.
-                                  for inds in np.split(potmatrix.indices, potmatrix.indptr[1:-1])], [buses]) * capacities
+        # See PyPSA-EUR paper (Hörsch et al. 2018, eq. (5), (7)) for further details
+
+        p_nom_max = capacities / (xr.where(
+            layoutmatrix.reindex_like(capacity_factor) != 0, capacity_factor,
+            0)).groupby('bus').max(['x', 'y']).fillna(0.)
     else:
-        raise AssertionError('Config key `potential` should be one of "simple" (default) or "conservative",'
-                             ' not "{}"'.format(p_nom_max_meth))
+        raise KeyError(
+            f'Config key `potential`={p_nom_max_meth} not implemented, should be "simple" (default) or "conservative",'
+        )
 
-    layout = xr.DataArray(np.asarray(potmatrix.sum(axis=0)).reshape(cutout.shape),
-                          [cutout.meta.indexes[ax] for ax in ['y', 'x']])
+    ## Determine weighted average distance from substation
 
-    # Determine weighted average distance from substation
-    cell_coords = cutout.grid_coordinates()
+    # Convert 2D to 1D: workaround to enable .nonzero() below
+    layoutmatrix = layoutmatrix.stack(spatial=('x', 'y'))
 
```

Comment:
Watch this, it might well be this stacking does not work. `cutout.grid_cells` and `cutout.grid_coordinates` are sorted in the same way as `cutotu.data[var].stack(spatial=['y', 'x'])` (see the [test function](https://github.com/PyPSA/atlite/blob/2fbdd9fa90739207c8bb53aa3c686d48e4b3f4cb/test/test_creation.py#L90))

Reply:
I did rewrite and test this piece with the rearranged index structure in mind.
It did and should work. Did you encounter problems?

Reply:
No that should be fine